truckersmp-cli

```diff
@@ -92,6 +94,39 @@ class AppId:
     proton = {}
 
 
+class DowngradeHTMLParser(html.parser.HTMLParser):
+    """Extract downgrade information from html."""
+
+    _data = {}
```

Comment:
I don't quite get what you're outlining here.

Presetting the values to 'false' is a good suggestion and I get that the calling function won't raise the error anymore and won't run into the exception anymore in this case. (Which would have return None)
Instead the 'if' will get a 'false' and continue on and run into the return at the end inside the try which returns None.
An error and a 'false' still both return None so the default answer is still "update to the latest version".

If you think we should catch the error further on and deny an update altogether we could also throw an exception and handle this in the calling function.

Reply:
That's right, but should we really depend on the availability of stats.truckersmp.com for upgrading games...? It might be that it's better to ask the user whether he or she wishes to upgrade on error.

Reply:
IMHO we should avoid waiting for user input. As a script it can be started by a .desktop file or from a launcher where we're unable to get an answer.

Reply:
To allow users to answer, let's add `--abort-update-if-supported-version-is-unknown` option :sweat_smile:

BTW what do you think of `sys.stdin.readline()` in `start_with_wine()`? Is it better to wait about a minute instead?

Reply:
> To allow users to answer, let's add `--abort-update-if-supported-version-is-unknown` option sweat_smile

And never document it :wink: 

> BTW what do you think of `sys.stdin.readline()` in `start_with_wine()`? Is it better to wait about a minute instead?

No, waiting just adds time and no one knows why. Two things would make sense here:
* Checking for pids of `steam.exe` and scan the environment for our prefix: [`tr '\0' '\n' < /proc/$(pidof steam.exe)/environ`](https://serverfault.com/a/66366)
* Checking within wine with the taskmanager of wine itself if steam is running: [`winedbg --command "info proc"`](https://stackoverflow.com/a/41227176)

Reply:
I forgot that we watch `loginusers.vdf` for Proton. How about watching `$WINEPREFIX/drive_c/Program Files (x86)/Steam/config/loginusers.vdf`?

Reply:
Should be also good.

Some thoughts:
* Installations don't have to be in the standard path or even inside the prefix, how do we know the location?
* I'm not sure how that behaves with different user at the same time since it's at a shared location. Would be a rare case anyway.
* We still have to look for steams process because a file change will only happen at login